database/models/core.py

```diff
@@ -303,6 +303,36 @@ def get_head_commit_notifications(self):
             )
         return []
 
+    def get_repository(self):
+        return self.repository
+
+    def get_commitid(self):
+        return None
+
+    @property
+    def external_id(self):
+        return self.pullid
+
+    @property
+    def id(self):
+        return self.id_
+
+    def should_write_to_storage(self) -> bool:
+        if self.repository is None or self.repository.owner is None:
+            return False
+        is_codecov_repo = self.repository.owner.username == "codecov"
+        return should_write_data_to_storage_config_check(
+            master_switch_key="pull_flare",
```

Comment:
I see this is referencing a `pull_flare` config value but is a generic method on the model.  I feel like maybe we chatted about this scenario in the past but what if there are multiple `ArchiveField`s on a single model?

Not a big deal right now but this might require a little reworking if that case does arise in the future.